sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -423,6 +423,45 @@ AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic
     size_t mqtt_topic_size,
     size_t* out_mqtt_topic_length);
 
+/**
+ * @brief Azure IoT Provisioning Client options for az_iot_provisioning_client_get_request_payload.
+ *        Not currently used.  Reserved for future use.
+ *
+ */
+typedef void az_iot_provisioning_client_payload_options;
+
+/**
+ * @brief Builds the optional payload for a provisioning request.
+ * @remark Use this API to build an MQTT payload during registration.
+ *         This call is optional for most scenarios.  Some service
+ *         applications may require #custom_payload_property specified during
+ *         registration to take additional decisions during provisioning time.
+ *         For example, if you need to register an IoT Plug and Play device you must
+ *         specify its model_id with this API via the #custom_payload_property
+ *         "{\"modelId\":\"" your_model_id "\"}".
+ *
+ * @param[in] client The #az_iot_provisioning_client to use for this call.
+ * @param[in] custom_payload_property __[nullable]__ Custom JSON to be added to this payload.
+ * Can be `NULL`.
+ * @param[in] reserved __[nullable]__ Reserved field for future options to this function.
+ * @param[out] mqtt_payload A buffer with sufficient capacity to hold the MQTT payload.
+ * @param[in] mqtt_payload_size The size, in bytes of \p mqtt_payload.
+ * @param[out] out_mqtt_payload_length __[nullable]__ Contains the string length, in bytes, of \p
+ * mqtt_payload. Can be `NULL`.
+ * @pre \p client must not be `NULL`.
+ * @pre \p reserved must be `NULL`
+ * @pre \p mqtt_payload must not be `NULL`.
+ * @pre \p mqtt_payload_size must be greater than 0.
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD az_result az_iot_provisioning_client_get_request_payload(
```

Comment:
nit: add a changelog entry for the new api

Reply:
@ahsonkhan - Yes.  

How do I do that, though?  The changelog is tied to releases whereas this will will just go in a "random" commit.  Is there some mechanism to track release IOU and when the release notes come out?

Reply:
You just add a `new feature` entry to the changelog markdown file here, for the "next" release:
https://github.com/Azure/azure-sdk-for-c/blob/main/CHANGELOG.md#120-beta2-unreleased

We try to add CL entries to that README in the same PR as the feature (or bug fix), and whenever we ship that particular version, those CL contents get auto-copied as part of the release notes (as part of CI/release process). The "iou" part is the fact that the next release is marked as "unreleased", until we decide to add a ship date to it and cut a release (which produces a git release and tag).

Reply:
Adding.